packages/lib-core/src/extensions/actions.ts

```diff
@@ -0,0 +1,119 @@
+import type { ReactNode } from 'react';
+import type { ExtensionHook, ExtensionK8sKindVersionModel } from '../api/common-types';
+import type { CodeRef, Extension } from '../types/extension';
+import type { AccessReviewResourceAttributes } from './core-types';
+
+/** ActionProvider contributes a hook that returns list of actions for specific context */
+export type ActionProvider = Extension<
```

Comment:
FYI @andrewballantyne our `Extension` type has two generic type parameters, as opposed to `Extension` type in Console (can't change the latter easily since it's already used by a lot of code).

This eliminates the need for `ExtensionDeclaration` type in this SDK.

Reply:
Right, we don't need two types because we have the ability to change the base `Extension` one :smiley: in other words, `Extension` with `type: string` in Console is a legacy bit that we can mitigate here.